### PR TITLE
Filter geoms outside panel

### PR DESCRIPTION
**Pull request by marcin-kam**

### CODE REVIEW

**Comment by jcubic** [comment]
Can you use GitHub username in commit? 
in repo:
```
git config --local user.email "you@eample.com"
```

File: R/contents.R [hunk 1]
```diff
@@ -106,6 +106,8 @@ unmapFactors <- function(df, origin) {
       if (length(origColumn) == nrow(df)) {
         # Simply add the column from the original data frame
         df[[name]] <- origColumn
+      } else {
+        df[[name]] <- origColumn[as.numeric(rownames(df))]
```
Comment (cosi1) [inline R/contents.R:110]: I prefer the following syntax (if possible):

```
variable <- if (condition) {
  value1
} else {
  value2
}
```

In my opinion, it improves readability and help avoid situations when nothing has been assigned to the variable at all (resulting in the variable not being initialized).

File: R/contents.R [hunk 2]
```diff
@@ -140,6 +142,49 @@ unmapAes <- function(data, mapping, plot) {
...
+    d
+  })
+}
+
+#' Get range data
+#'
+#' Depends on ggplot2 version
+#'
+getRanges <- function(plot, built) {
+  if (packageVersion("ggplot2")$major == 2L) {
```
Comment (cosi1) [inline R/contents.R:166]: I think we should provide a unified function for checking that...

File: R/contents.R [hunk 3]
```diff
@@ -299,6 +344,7 @@ removeRowsWithNA <- function(data, mapping, layers) {
 getTooltipData <- function(plot, built, varDict, plotScales, callback) {
   mapping <- getLayerAesthetics(plot)
   data <- built$data
+  data <- removeOutOfRangeData(data, plot, built)
```
Comment (cosi1) [inline R/contents.R:347]: Please try to use named arguments when calling your custom functions – the order of arguments may change in the future!

**Review by cosi1 — APPROVED** [review]
LGTM, but please see my comments before merging.

**Pull request merged**